Add a Code of Conduct

MarkBennett opened the pull request:
As part of opening our Slack, we've decided that we need a Code of Conduct.

See devedmonton/devedmonton-org#22 for details.

MarkBennett committed:
Github autogenerated a Code of Conduct

What?!

MarkBennett committed:
Add a weapons policy and tweak for Slack

MarkBennett committed:
Link to Alberta Human Rights Act

ashiazed commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,90 @@
+
+We invite all those who participate in Dev Edmonton Society to help us create safe and positive experiences for everyone.
+
+## 2. Open Source Citizenship
```
Should we change this to Open Culture Citizenship :thinking:

ashiazed committed:
Swap open source to open culture

More relevant to our goals and community

ashiazed commented:
I'm not sure if we need a tldr but I like the ExchangeJS one

ashiazed commented:
There is a link to a coc-report page in the COC, do you want to create that in this PR?
https://www.djangoproject.com/conduct/reporting/ Is probably the best one to work off of.

MarkBennett commented:
> There is a link to a coc-report page in the COC, do you want to create that in this PR?
> https://www.djangoproject.com/conduct/reporting/ Is probably the best one to work off of.

Yes. I was looking for some guidelines and a good one to follow as I've got no experience with this. That example looks great! I'll add it to the PR.

MarkBennett committed:
Shamelessly swipe this from Django

Still need to update this with our guidelines.

MarkBennett committed:
Update reporting links

MarkBennett commented:
I found the source for the Djange reporting policy and cut-and-paste it into this branch:

https://github.com/i3/i3.github.io/blob/master/conduct-reporting.html

We'll need to read through and change it so that it makes sense for our organization.

/cc @ashiazed

MarkBennett committed:
Switch to Markdown and use Dev Edmoton Society

Drop references to i3

MarkBennett commented:
I've updated the CoC reporting that @ashiazed found to use our names and email.

Email forwarding is setup to go to me via Mailgun and should start working in the next 24-48.

MarkBennett committed:
Dropped grievance process

We don't have a clear way to do this yet, and we're not even sure they'll be need. Holding off on this until we've got more experience with the society.

MarkBennett committed:
Fix heading numbers and contact email

MarkBennett commented:
Will close this issue end of day on Sunday September 30th. The text of this Code of Conduct will then be circulated as a board resolution. Please get in any changes or feedback before then. Thanks!

benzittlau commented on CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,84 @@
+
+## 1. Purpose
+
+A primary goal of Dev Edmonton Society is to be inclusive to the largest number of contributors, with the most varied and diverse backgrounds possible. As such, we are committed to providing a friendly, safe and welcoming environment for all, regardless of gender, sexual orientation, ability, ethnicity, socioeconomic status, religion (or lack thereof) and any personal characteristics protected un ...
```
Is contributors the right noun for us?

benzittlau replied:
Maybe members or community members?

MarkBennett replied:
@benzittlau I'm not 100% sold on it, but wasn't really sure what else to use. Members isn't right as not all people engaging with us are members. Examples of people this would apply to are:

* meetup and event attendees
* Slack users
* meetup organizers
* board members

`People` might work? As in, `A primary goal of Dev Edmonton Society is to be inclusive to the largest number of people`. It's also in our mission which is a nice symmetry.

benzittlau replied:
I noticed "community members" is used elsewhere in the document, any concerns with that?

MarkBennett replied:
> I noticed "community members" is used elsewhere in the document, any concerns with that?

Can you tell that I copied this extensively from the PythonConf CoC? I'll update those too.

benzittlau commented on coc_reporting.md:
```diff
@@ -0,0 +1,49 @@
+Largely taken from [django](https://www.djangoproject.com/conduct/reporting/). Thanks!
+
+If you believe someone is violating the code of conduct we ask that you report it to the Dev Edmonton Society Board by emailing conduct AT devedmonton DOT org. **All reports will be kept confidential.** In some cases we may determine that a public statement will need to be made. If that's the case, the identities of all victims and reporters will remain confidential unless those individuals instr ...
+
```
What are our views on anonymous reports?  Having looked at this in the past it's harder than you'd think to create a "dropbox" that's actually anonymous.  In particular if you want to have a further conversation with the reporter for clarification, etc. to have an anonymous platform that allows for 2-way conversation is more involved.  Not sure if it's worth worrying about, but it always makes me a little uncomfortable to require people to identify themselves in order to report.

MarkBennett replied:
Good thoughts @benzittlau. @Estyn had some questions about this too. Here's my take...

If the mission of our code of conduct and the reporting process is to identify and address issues that undermine the culture and diversity of our community, then there are some pros and cons to anonymous reporting. 

### Pros
* Ensure that people don't hesitate to report issues
* Ensures that reporters identities aren't released by us (as we don't know them!)

### Cons
* Without being able to identify individuals, the scope and volume of issues is unknown
* Truly "anonymous" reporting is very hard to setup
* To act on a situation, and ensure it's resolved, we may require more information and context which anonymous reports don't provide
* There may be speculation about the source of anonymous reporters
* Though unlikely, anonymous reporting allows bad actors to disrupt the community

By building a firewall within our organization, we can ensure that only parties directly involved in a report are privy to the details and identities of those involved. We *need* to convince people in our community that we take reporter privacy seriously, and understand the personal and professional risks they may be taking. Otherwise, people may be less inclined to participate and we fail in our mission of improving diversity and inclusiveness.

I think we could address this a bit better by indicating that Dev Edmonton won't ever share reporter identities with the public, or with the individual a grievance is against. This includes restricting access to information that could be used to identify a reporter (like dates/times, names, phone numbers, emails, aliases, etc.) That said we will share the report with a subset of the board required to address it properly.

I'll try to update the language in the reporting page and CoC to make this clearer.

benzittlau commented:
Other than the one wording question, and the question around anonymous reports, looks good to me!

Estyn commented:
Looks good to me.  Personally I'm fine with allowing anonymous reports, but I'm not sure it's something that we need to explicitly state in the document.

abramhindle committed:
Fix a hyperlink to alberta human rights act

MarkBennett commented:
> Looks good to me. Personally I'm fine with allowing anonymous reports, but I'm not sure it's something that we need to explicitly state in the document.

@Estyn Ben had a similar comment. I addressed it here...

https://github.com/devedmonton/devedmonton.github.io/pull/2/files#r221007919

abramhindle commented:
Regarding scope:

> This code of conduct and its related procedures also applies to unacceptable behavior occurring outside the scope of community activities when such behavior has the potential to adversely affect the safety and well-being of community members.

This broadens the scope of the code of conduct very far. It is apparent is it supposed to be used against disgraced individuals who bring shame to the community through external behaviour and actions. But what if it is used against say someone who supports a right wing party that seeks to infringe on the rights of citizens. We should not exclude faith groups, conservatives, extreme leftists from joining if they abide by the code of conduct.

I fear this scope is too far reaching and not properly scoped---it is as if there is no scope whatsoever. I don't want to be excluding people based on politcal beliefs unless their beliefs are genocidal. If someone holds certain views on various sides of the political spectrum and are not expressing in community interaction, yet express them outside of community interaction and someone finds this offensive, they might make a safety argument.

Even the wording is vague:
> has the potential to adversely affect the safety and well-being of community members

This sounds like an avenue to allow a political purge from the society of members who align with pretty commonly held beliefs in mainstream society.

my suggestion

> This code of conduct and its related procedures also applies to unacceptable behavior occurring outside the scope of community activities when **such behavior adversely affects** the safety **or** well-being of community members.

This minor change might be enough to allow sex workers who are developers to not infringe on:

>  *   Posting or displaying sexually explicit or violent material. 

Or members from political or faith groups who might violate:

> *   Sexist, racist, homophobic, transphobic, ableist or otherwise discriminatory jokes and language.

yet do not so in the context of the community itself.

This would allow us to deal with out-right genocide promoters as it would harm the society to be associated with them, but it would allow people who are part of relatively well respected groups in public discourse to still take part in the society without breaching the code of conduct.

Please don't read too much into my intentions, I simply want to avoid the use of the CoC to exclude people who espouse beliefs---outside of the community context---that are still acceptable in the mainstream political discourse. And this argument is more about fairness and inclusiveness. It's hard to make this kind of suggestion without putting your foot in your mouth and running afoul of others.

abramhindle commented:
Through private discussion with board members I do not believe my suggested changes would be effective at achieving the goals of appropriately scoping the reach of the CoC and it is best left to the board to decide on a case by case basis. Feel free to ignore my minor revision, but board members should be wary of scope overreach.

MarkBennett committed:
Rename contributors to people

MarkBennett committed:
Clarify that we prefer education and communication..

..but that temporary and permanent bans are on the table should violations and bad behavior continue.

SteveReiter commented:
Looks good to me.

MarkBennett commented:
This has gone our for approval in board resolution #0007. You can vote here if you're a board member:

https://goo.gl/forms/t39tqrAah5oafAbF2

nathany commented on coc_reporting.md:
```diff
@@ -0,0 +1,49 @@
+
+Largely taken from [django](https://www.djangoproject.com/conduct/reporting/). Thanks!
+
+If you believe someone is violating the code of conduct we ask that you report it to the Dev Edmonton Society Board by emailing conduct AT devedmonton DOT org. **All reports will be kept confidential.** In some cases we may determine that a public statement will need to be made. If that's the case, the identities of all victims and reporters will remain confidential unless those individuals instr ...
```
Is this email address set up?

MarkBennett replied:
Yes.

MarkBennett committed:
Rename CODE_OF_CONDUCT.md to code_of_conduct.md

MarkBennett commented:
As per the board resolution 00007, I'm merging the approved Code of Conduct.
